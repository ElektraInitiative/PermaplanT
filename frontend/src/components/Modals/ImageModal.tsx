import TransparentBackground from '../TransparentBackground';
import ModalContainer from './ModalContainer';
import { ReactNode } from 'react';

interface ImageModalProps {
  /** Modal headline displayed on top.*/
  title: string;
  /** The main element of the modal, usually an image.*/
  body: ReactNode;
  /** Callback that informs the parent when the modal should be hidden/displayed (e.g. when the user pressed the close button).*/
  setShow: (show: boolean) => void;
  /** Decides whether the modal should be shown.*/
  show: boolean;
  /** Click callback for cancel/close button.*/
  onCancel: () => void;
}

/**
 * A modal component used to display images.
 * @param props.title Modal headline displayed on top.
 * @param props.body The main element of the modal, usually an image.
 * @param props.setShow Callback that informs the parent when the modal should be hidden/displayed (e.g. when the user pressed the close button).
 * @param props.show Decides whether the modal should be shown.
 * @param props.onCancel Click callback for cancel/close button.
 */
export default function ImageModal({ title, body, setShow, show, onCancel }: ImageModalProps) {
  return (
    <>
      <TransparentBackground
        onClick={() => {
          setShow(false);
        }}
        show={show}
      />
      <ModalContainer show={show}>
        <div className="flex h-[70vh] w-[80vw] flex-col rounded-lg bg-neutral-100 p-6 dark:bg-neutral-100-dark">
          <div className="flex justify-between">
            <h2>{title}</h2>
            <button
              onClick={onCancel}
              className="rounded-lg border border-neutral-600 px-5 py-2.5 text-center text-sm font-medium hover:bg-neutral-600 focus:outline-none focus:ring-4 focus:ring-blue-300 sm:w-auto"
            >
              x
            </button>
          </div>
          <div className="flex justify-center w-full h-full max-h-[60vh] p-4">
            {body}
          </div>
        </div>
      </ModalContainer>
    </>
  );
}
