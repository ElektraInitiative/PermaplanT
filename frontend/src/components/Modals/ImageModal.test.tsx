import ImageModal from './ImageModal';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const callback = jest.fn();
  const tree = renderer
    .create(
      <MemoryRouter>
        <ImageModal title="Test" body="Test" setShow={callback} show={true} onCancel={callback} />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
