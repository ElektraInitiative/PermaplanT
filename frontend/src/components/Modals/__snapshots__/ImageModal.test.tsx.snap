// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
[
  <div
    className="fixed bottom-0 right-0 z-[1010] "
    onClick={[Function]}
  >
    <div
      style={
        {
          "opacity": 0,
        }
      }
    >
      <div
        className="h-screen w-screen bg-[#000]"
      />
    </div>
  </div>,
  <div
    className="fixed left-[50%] top-[15%] z-[1020]"
    style={
      {
        "opacity": 0,
        "transform": "translateX(-50%) translateZ(0)",
      }
    }
  >
    <div
      className="flex h-[70vh] w-[80vw] flex-col rounded-lg bg-neutral-100 p-6 dark:bg-neutral-100-dark"
    >
      <div
        className="flex justify-between"
      >
        <h2>
          Test
        </h2>
        <button
          className="rounded-lg border border-neutral-600 px-5 py-2.5 text-center text-sm font-medium hover:bg-neutral-600 focus:outline-none focus:ring-4 focus:ring-blue-300 sm:w-auto"
          onClick={[MockFunction spy]}
        >
          x
        </button>
      </div>
      <div
        className="flex h-full max-h-[60vh] w-full justify-center p-4"
      >
        Test
      </div>
    </div>
  </div>,
]
`;
